drivers/sensors/bme680.c

```diff
@@ -0,0 +1,1764 @@
+ *   @devno - The user-specified device number, starting from 0
+ *   @i2c   - An instance of the I2C interface to use to communicate with
+ *           BME680
+ *
+ * Return value:
+ *   Zero (OK) on success; a negated errno value on failure
+ ****************************************************************************/
+
+int bme680_register(int devno, FAR struct i2c_master_s *i2c)
+{
+  FAR struct sensor_lowerhalf_s *lower;
+  FAR struct bme680_dev_s *priv;
+  FAR char *argv[2];
+  char arg1[32];
+  int ret = OK;
+
+  DEBUGASSERT(i2c != NULL);
+
+  /* Initialize the BME680 device structure */
+
+  priv = (FAR struct bme680_dev_s *)kmm_zalloc(sizeof(struct bme680_dev_s));
+  if (priv == NULL)
+    {
+      snerr("ERROR: Failed to allocate instance (err = %d)\n", ret);
+      return -ENOMEM;
+    }
+
+  priv->i2c = i2c;
+  priv->enabled = false;
+  nxmutex_init(&priv->dev_lock);
+  nxsem_init(&priv->run, 0, 0);
+
+  /* Check Device ID */
+
+  ret = bme680_checkid(priv);
+  if (ret < 0)
+    {
+      snerr("ERROR: Wrong device ID!\n");
+      goto err_init;
+    }
+
+  /* Get Calibration Data */
+
+  ret = bme680_get_calib_data(priv);
+
+  if (ret < 0)
+    {
+      snerr("Failed to read calib data from bme680:%d\n", ret);
+      goto err_init;
+    }
+
+#ifndef CONFIG_BME680_DISABLE_PRESS_MEAS
+  /* Register the barometer driver */
+
+  lower = &priv->dev.lower[BME680_PRESS_IDX];
+  lower->ops = &g_sensor_ops;
+  lower->type = SENSOR_TYPE_BAROMETER;
+
+  ret = sensor_register(lower, devno);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to register barometer \
+              driver (err = %d)\n",
+            ret);
+      goto err_init;
+    }
+#else
+  /* Register the temperature driver */
+
+  lower = &priv->dev.lower[BME680_TEMP_IDX];
+  lower->ops = &g_sensor_ops;
+  lower->type = SENSOR_TYPE_AMBIENT_TEMPERATURE;
+
+  ret = sensor_register(lower, devno);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to register temperature \
+              driver (err = %d)\n",
+            ret);
+      goto err_init;
+    }
+#endif
+
+#ifndef CONFIG_BME680_DISABLE_HUM_MEAS
+  /* Register the humidity driver */
+
+  lower = &priv->dev.lower[BME680_HUM_IDX];
+  lower->ops = &g_sensor_ops;
+  lower->type = SENSOR_TYPE_RELATIVE_HUMIDITY;
+
+  ret = sensor_register(lower, devno);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to register humidity \
+              driver (err = %d)\n",
+            ret);
+      goto err_init;
+    }
+#endif
+
+#ifndef CONFIG_BME680_DISABLE_GAS_MEAS
+  /* Register the gas driver */
+
+  lower = &priv->dev.lower[BME680_GAS_IDX];
+  lower->ops = &g_sensor_ops;
+  lower->type = SENSOR_TYPE_GAS;
+
+  ret = sensor_register(lower, devno);
+  if (ret < 0)
+    {
+      snerr("ERROR: Failed to register gas \
+              driver (err = %d)\n",
+            ret);
+      goto err_init;
+    }
+#endif
+
+  /* Create thread for polling sensor data */
+
+  snprintf(arg1, 16, "0x%" PRIxPTR, (uintptr_t)priv);
```

Comment:
```suggestion
  snprintf(arg1, 16, "%p", priv);
```
like the patch: https://github.com/apache/nuttx/pull/9887